config/config_test.go

```diff
@@ -0,0 +1,168 @@
+package config
```

Comment:
This is really interesting, and I've added this to my Evernote. I especially like the philosophy of working backward from the actual user experience.

When I started doing TDD, I noticed I wrote my code differently, to be more easily testable. But I guess if you have that in mind as you code in the first place, maybe you don't need TDD to get that benefit. It does put a bit more of an onus to discipline yourself to add test coverage after the fact, but I still like the focus on UX.

Thanks for sharing.